packages/react/src/QuestionnaireBuilder/QuestionnaireBuilder.tsx

```diff
@@ -665,3 +736,21 @@ function createPage(): QuestionnaireItem {
     ],
   } as QuestionnaireItem;
 }
+
+function reorderItems(
+  items: QuestionnaireItem[] | undefined,
+  itemToMove: QuestionnaireItem,
+  delta: number
+): QuestionnaireItem[] | null {
```

Comment:
We almost never return `null` from any of our methods, except in very rare circumstances.  Usually we'd return `undefined` for a "nothing" result.

However, in this case, I think we should always return the `QuestionnaireItem[]` array.  If the delta is invalid, then we can just return the original array.  Fewer edges cases to worry about (note that we're not checking for `null` return values at the call site).

As mentioned above, I think we should change `itemToMove` to `itemIndex` and pass in the numeric array index.  That should avoid the `findIndex` call.

Then you can remove the `index === -1` check.

```ts
const newIndex = itemIndex + delta;
if (newIndex < 0 || newIndex >= items.length) {
  return items;
}
...
```